File: app/models/conceptoingreso.rb
class Conceptoingreso < ActiveRecord::Base
		self.table_name = "concepto_ingresos"
    has_many :ingreso
   def grabar(datos)
     @objconceptoingreso = Conceptoingreso.new
   
   valor = 0
    if @objconceptoingreso!=nil

      @objconceptoingreso.codigo_concepto_ingreso = datos[0].to_s
      @objconceptoingreso.nombre = datos[1].to_s
      @objconceptoingreso.status = datos[2].to_s
      
      @objconceptoingreso.save
      $tirajson = '{ "success": "true", "msg": "Datos guardados satisfactoriamente!" }'
      valor = 1

    end 
   return valor
  end

    def buscar()
 
       @objconceptoingreso = Conceptoingreso.all
   @son = Conceptoingreso.count
   if @son > 0 
    @i=1
    tirajson = '[  '
    @objconceptoingreso.each do |conceptosingresos|
     if @i<@son
      tirajson = tirajson +   ' { "id": "'        + conceptosingresos.id.to_s +
                              '", "nombre": "'        + conceptosingresos.nombre.to_s+ 
                              '", "codigo": "'   + conceptosingresos.codigo_concepto_ingreso.to_s +
                            '", "status": "'      + conceptosingresos.status.to_s + '"}, '                              
     else
      tirajson = tirajson +   ' { "id": "'        + conceptosingresos.id.to_s +
                              '", "nombre": "'        + conceptosingresos.nombre.to_s + 
                              '", "codigo": "'   + conceptosingresos.codigo_concepto_ingreso.to_s+
                              '", "status": "'      + conceptosingresos.status.to_s + '"} '     
     end
     @i=@i+1
    end
    tirajson = tirajson + ' ]'
   else
    tirajson = '{ "success": "true", "exito": "false", "msg": "No hay datos!" }'; 
   end
   return tirajson 
  



  end
end

File: app/assets/javascripts/Siaco/view/ConceptoIngreso.js
Ext.onReady(function() {
	
	Ext.QuickTips.init();
	Ext.define('Siaco.view.MiPanelConceptoingresos', {
		extend: 'Ext.form.Panel',
		
		initComponent : function() {
			this.dockedItems = [{
				dock: 'bottom',
				buttons: [{
					text: 'Limpiar',
					iconCls:'limpiar',
					handler: function() {
						Ext.getCmp('conceptoingreso').getForm().reset();
					}
					
				},{
				text: 'Grabar',
				iconCls:'grabar',
				handler:function() {
					guardarconceptoingreso();
				}
				},{
				text: 'Eliminar',
				iconCls:'eliminar',
				handler:function() {
					Ext.Msg.alert('Pronto','En construccion');
				}
				}]
			}];
			this.callParent();
		}
	});

Ext.define('Siaco.view.ConceptoIngreso', {
		extend: 'Siaco.view.MiPanelConceptoingresos',
		xtype: 'conceptoingresoView',
		alias: 'widget.conceptoingreso',
		id: 'conceptoingreso',
		title: 'NuevoConceptoIngreso',
		bodyPadding: 5,
		width: 450,
		
		initComponent: function () {
			
			this.items = [
			{
				fieldLabel: 'Nombre',
				xtype: 'textfield',
				name: 'nombre',
				id: 'nombreconceptoingreso'
			},{
				fieldLabel: 'Codigo',
				xtype: 'textfield',
				name: 'codigo',
				id: 'codigoconceptoingreso'
			}
			];
			this.callParent();
			Ext.getCmp('nombreconceptoingreso').focus();
		}
	});
});

   function guardarconceptoingreso()
   {
   		Ext.Ajax.request({
	     url: 'conceptosingresos/grabar',
		 method: 'GET',
	     //Enviando los parametros a la pagina servidora
	     params: {
	      ajax: 'true',
	      funcion: 'grabar',

	      nombre: Ext.getCmp('nombreconceptoingreso').getValue(), //obtiene el valor a traves del id del campo
	      codigo: Ext.getCmp('codigoconceptoingreso').getValue(),
	     },
	     //Retorno exitoso de la pagina servidora a traves del formato JSON
	     success: function( resultado, request ) {
	      datos=Ext.JSON.decode(resultado.responseText);
	      Ext.Msg.alert('Exito', datos.msg);
	      Ext.getCmp('conceptoingreso').getForm().reset();
	     },
	     //No hay retorno de la pagina servidora
	     failure: function() {
	      Ext.Msg.alert("Error", "Servidor no conectado!");
	     }
	    });
   }